fastcov.py

```diff
@@ -151,35 +151,55 @@ def processGcovs(cwd, gcov_files, gcov_filter_options):
     return files
 
 def dumpBranchCoverageToLcovInfo(f, branches):
-    branch_miss  = 0
+    branch_miss = 0
+    branch_found = 0
+    brda = []
     for line_num, branch_counts in branches.items():
         for i, count in enumerate(branch_counts):
-            #Branch (<line number>, <block number>, <branch number>, <taken>)
-            f.write("BRDA:{},{},{},{}\n".format(line_num, int(i/2), i, count))
+            # Branch (<line number>, <block number>, <branch number>, <taken>)
+            brda.append((int(line_num), int(i/2), i, int(count)))  # XXX: just use int for line_num/count throughout
             branch_miss += int(count == 0)
-    f.write("BRF:{}\n".format(len(branches)))                 #Branches Found
-    f.write("BRH:{}\n".format((len(branches) - branch_miss))) #Branches Hit
+            branch_found += 1
+    for v in sorted(brda):
+        f.write("BRDA:{},{},{},{}\n".format(*v))
+    f.write("BRF:{}\n".format(branch_found))                # Branches Found
+    f.write("BRH:{}\n".format(branch_found - branch_miss))  # Branches Hit
 
 def dumpToLcovInfo(fastcov_json, output):
     with open(output, "w") as f:
-        for sf, data in fastcov_json["sources"].items():
+        sources = fastcov_json["sources"]
+        for sf in sorted(sources.keys()):
+            data = sources[sf]
             f.write("SF:{}\n".format(sf)) #Source File
 
             fn_miss = 0
+            fn = []
+            fnda = []
             for function, fdata in data["functions"].items():
-                f.write("FN:{},{}\n".format(fdata["start_line"], function))          #Function Start Line
-                f.write("FNDA:{},{}\n".format(fdata["execution_count"], function))   #Function Hits
+                fn.append((fdata["start_line"], function))  # Function Start Line
+                fnda.append((fdata["execution_count"], function))  # Function Hits
                 fn_miss += int(fdata["execution_count"] == 0)
+            # NOTE: lcov sorts FN, but not FNDA.
+            for v in sorted(fn):
+                f.write("FN:{},{}\n".format(*v))
+            for v in sorted(fnda):
+                f.write("FNDA:{},{}\n".format(*v))
             f.write("FNF:{}\n".format(len(data["functions"])))               #Functions Found
             f.write("FNH:{}\n".format((len(data["functions"]) - fn_miss)))   #Functions Hit
 
             if data["branches"]:
                 dumpBranchCoverageToLcovInfo(f, data["branches"])
 
             line_miss = 0
+            da = []
             for line_num, count in data["lines"].items():
-                f.write("DA:{},{}\n".format(line_num, count)) #Line
+                # NOTE: lcov turns this into ints, fastcov keeps floats here
+                # (e.g. "DA:119,525694020.0" - for large numbers in header
+                # files?!)
```

Comment:
Can you explain what you mean by "fastcov keeps floats here"? As far as I know fastcov is only using integers for counts, not sure how a decimal is being formatted in

Reply:
Yes.  It seemed like for bigger values it would get floats there (but always just `.0`).
Have not investigated where it is coming from, but would be gcov's JSON then I suppose.